### PR TITLE
Error layer

### PR DESCRIPTION
**Pull request by maggie97**

### CODE REVIEW

File: teckers/teckers/HomeController.swift [hunk 1]
```diff
@@ -77,6 +77,19 @@ class HomeController: UIViewController {
         let messageCell = MessageCell(style: .default, reuseIdentifier: "")
         messageTableView.register(messageCell.nibName , forCellReuseIdentifier: messageCell.NameCell)
     }
+    override func viewWillAppear(_ animated: Bool) {
+        NetworkError.delegate = self
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        if let view = UIApplication.shared.windows[0].rootViewController?.children[0] as? LoginViewController{
+            if let info = NetworkError.instance.getInformation(for: 401){
```
Comment (cortestavo) [inline teckers/teckers/HomeController.swift]: avoid to use raw values, use an enum instead

File: teckers/teckers/HomeController.swift [hunk 2]
```diff
@@ -115,3 +128,17 @@ extension HomeController : UISearchBarDelegate {
...
+    func dismiss() {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func goTo(with segueIdentifier: Segues) {
+        self.performSegue(withIdentifier: segueIdentifier.rawValue, sender: nil)
+    }
+    
+    func error(message : String){
+        let alertNotification = Alert(title: "Error", massage: message, type: 0)
```
Comment (cortestavo) [inline teckers/teckers/HomeController.swift]: type doesn't exist anymore

File: teckers/teckers/NetworkError.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                         .unauthorized : (TypesNetworkErrors.unauthorized.rawValue, 0),
+                         .forbidden: (TypesNetworkErrors.forbidden.rawValue, 0),
+                         .pageNotFound : (TypesNetworkErrors.pageNotFound.rawValue, 0),
+                         .requestTimeOut : (TypesNetworkErrors.requestTimeOut.rawValue, 1),
+                         .noInternet: (TypesNetworkErrors.noInternet.rawValue, 0),
+                         .noJSON: (TypesNetworkErrors.noJSON.rawValue, 0),
+                         .internalServerError: (TypesNetworkErrors.internalServerError.rawValue, 0)]
+        self.actionsBeforeAlert = [.unauthorized: {
+            DispatchQueue.main.async{
+//                NetworkError.delegate?.dismiss()
```
Comment (cortestavo) [inline teckers/teckers/NetworkError.swift]: remove comment

File: teckers/teckers/NetworkError.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            }
+        }]
+    }
+
+     func getInformation( for statusCode: Int) ->  (message: String, type: Int)?{
```
Comment (cortestavo) [inline teckers/teckers/NetworkError.swift]: create an enum for http responses

File: teckers/teckers/NetworkError.swift [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            return messageErrors[.internalServerError]
+        case 600:
+            return messageErrors[.noInternet]
+        default:
+            return messageErrors[.unknow]
+        }
+    }
+    func getActionBeforeAlert( for statusCode: Int) ->  (() -> Void)? {
+        switch statusCode{
+        case 401:
```
Comment (cortestavo) [inline teckers/teckers/NetworkError.swift]: use enum case instead of int hardcoded value

File: teckers/teckers/NetworkError.swift [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            return actionsBeforeAlert[.unauthorized]
+        default:
+            return nil
+        }
+    }
+
+     func getAction(for statusCode: Int) -> () -> Void {
```
Comment (cortestavo) [inline teckers/teckers/NetworkError.swift]: use an enum case instead an int value

File: teckers/teckers/NetworkHandler.swift [hunk 7]
```diff
@@ -15,12 +15,27 @@ struct NetworkHandler {
     }
     
     static func request(url : URLRequestConvertible, onSucess success : @escaping (_ JSON : [String : Any]) -> Void,  onFailure failure: @escaping (_ error: Error) -> Void){
+        if !isConnected(){
+            NetworkError.instance.getAction(for: 600)()
```
Comment (cortestavo) [inline teckers/teckers/NetworkHandler.swift]: use an enum case instead an int value

File: teckers/teckers/NetworkHandler.swift [hunk 8]
```diff
@@ -15,12 +15,27 @@ struct NetworkHandler {
...
+            return
+        }
         Alamofire.request(url).responseJSON{ response in
-            if let Error = response.error {
-                failure(Error)
+            guard let statusCode = response.response?.statusCode else{
+                NetworkError.instance.getAction(for: -1)()
```
Comment (cortestavo) [inline teckers/teckers/NetworkHandler.swift]: use an enum case instead an int value

File: teckers/teckers/NetworkHandler.swift [hunk 9]
```diff
@@ -15,12 +15,27 @@ struct NetworkHandler {
...
+                return
             }
-            else if let jsonResponseBackend = response.value as? [String:Any] {
-                success(jsonResponseBackend)
+            switch response.result{
+            case .success(let value):
+                switch statusCode {
+                case 100..<400:
```
Comment (cortestavo) [inline teckers/teckers/NetworkHandler.swift]: verify if this case is necessary

**Review by cortestavo — CHANGES_REQUESTED** [review]

**Review by chirica2501 — APPROVED** [review]

**Pull request merged**